### ISSUE TITLE
Error setting up R environment

### ISSUE THREAD

**Issue by usanchez** [onnela-lab/Beiwe-Analysis#6]
Hi,

I was trying to set up the local environment of the Data Pipeline, so I could test everything before deploying it on the cloud. The problem is that when trying to install the R requirements, I get the following errors:

```
Installing package into ‘/home/ubuntu/R/x86_64-pc-linux-gnu-library/3.4’
(as ‘lib’ is unspecified)
Warning: dependency ‘cowplot’ is not available
trying URL 'https://cran.us.r-project.org/src/contrib/ggpubr_0.2.4.tar.gz'
Content type 'application/x-gzip' length 2665439 bytes (2.5 MB)
==================================================
downloaded 2.5 MB

ERROR: dependency ‘cowplot’ is not available for package ‘ggpubr’
* removing ‘/home/ubuntu/R/x86_64-pc-linux-gnu-library/3.4/ggpubr’

The downloaded source packages are in
        ‘/tmp/RtmprfcfL5/downloaded_packages’
Installing package into ‘/home/ubuntu/R/x86_64-pc-linux-gnu-library/3.4’
(as ‘lib’ is unspecified)
Warning: dependencies ‘DescTools’, ‘mvtnorm’ are not available
also installing the dependencies ‘libcoin’, ‘multcomp’, ‘coin’

trying URL 'https://cran.us.r-project.org/src/contrib/libcoin_1.0-5.tar.gz'
Content type 'application/x-gzip' length 779010 bytes (760 KB)
==================================================
downloaded 760 KB

trying URL 'https://cran.us.r-project.org/src/contrib/multcomp_1.4-11.tar.gz'
Content type 'application/x-gzip' length 603504 bytes (589 KB)
==================================================
downloaded 589 KB

trying URL 'https://cran.us.r-project.org/src/contrib/coin_1.3-1.tar.gz'
Content type 'application/x-gzip' length 1263884 bytes (1.2 MB)
==================================================
downloaded 1.2 MB

trying URL 'https://cran.us.r-project.org/src/contrib/rcompanion_2.3.7.tar.gz'
Content type 'application/x-gzip' length 113737 bytes (111 KB)
==================================================
downloaded 111 KB

ERROR: dependency ‘mvtnorm’ is not available for package ‘libcoin’
* removing ‘/home/ubuntu/R/x86_64-pc-linux-gnu-library/3.4/libcoin’
ERROR: dependency ‘mvtnorm’ is not available for package ‘multcomp’
* removing ‘/home/ubuntu/R/x86_64-pc-linux-gnu-library/3.4/multcomp’
ERROR: dependencies ‘libcoin’, ‘mvtnorm’, ‘multcomp’ are not available for package ‘coin’
* removing ‘/home/ubuntu/R/x86_64-pc-linux-gnu-library/3.4/coin’
ERROR: dependencies ‘DescTools’, ‘coin’ are not available for package ‘rcompanion’
* removing ‘/home/ubuntu/R/x86_64-pc-linux-gnu-library/3.4/rcompanion’

The downloaded source packages are in
        ‘/tmp/RtmprfcfL5/downloaded_packages’
Warning messages:
1: In install.packages(p, repos = "https://cran.us.r-project.org") :
  installation of package ‘ggpubr’ had non-zero exit status
2: In install.packages(p, repos = "https://cran.us.r-project.org") :
  installation of package ‘libcoin’ had non-zero exit status
3: In install.packages(p, repos = "https://cran.us.r-project.org") :
  installation of package ‘multcomp’ had non-zero exit status
4: In install.packages(p, repos = "https://cran.us.r-project.org") :
  installation of package ‘coin’ had non-zero exit status
5: In install.packages(p, repos = "https://cran.us.r-project.org") :
  installation of package ‘rcompanion’ had non-zero exit status
```

Which version of R was developed everything on? Could we have the specs of the R development environment? 

Thank you very much.

Best wishes, 

Unai.

**Comment by usanchez** [comment]
I think that I should dig up in the code and the errors a little bit more before creating an issue, but anyway, I hope that it may be useful for anyone out there.

The problem was that by running `apt install r-core-base littler` to install the `R` language and the command line `R` version, we get (if I remember correctly) the version `3.4` for `R`. However, the version needed for the `Pipeline` to work is `3.5.2`. So, I dig up in the code and found the answer in the [`Dockerfile`](https://github.com/onnela-lab/beiwe-backend/blob/f3139bbc4932393e1ba7279ddccf3bcb421dbd3a/pipeline/Dockerfile) from the `beiwe-backend` repo. This file is the one that sets up the container in the cloud, and so it installs R to be able to run all the code. So, I converted the `Docker` instructions into commands for my ubuntu 18, and got the following commands (you should run them from `home`):

```
# Install system requirements
sudo apt update
sudo apt install -y gnupg2 python3-pip software-properties-common git

# Ubuntu 18.04 repository doesn't include R(3.5.2). To fix this we need to add
# an external repository to install R from. First, we need to add GPG key:
sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9
# Once we have the trusted key, we add the repository as follows:
sudo add-apt-repository "deb https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/"
sudo apt update
sudo apt install -y r-base

# Install python requirements (in case you need to run it)
sudo pip3 install --upgrade pip
sudo pip3 install -r Beiwe-Analysis/Pipeline/python3-requirements.txt

# Install R requirements
sudo Rscript --vanilla Beiwe-Analysis/Summary/install_required_packages.R
```

Note that should already have by this point `pip3` and `git` installed to be able to clone the repo and install the `python3` requirements.

**Comment by usanchez** [comment]
I could set up the environment correctly and the R code worked perfectly with these steps, so I think I'll close this issue.

**Comment by biblicabeebli** [comment]
Hi Unai,
Can you clarify what the underlying issues here was?  I'm reading your report as an inability to install R requirements on your _local_ setup, most likely due to running a different version of ubuntu?

**Comment by usanchez** [comment]
Hi Eli,

As stated by the steps in the instructions for the Data Pipeline, I was setting up the environment to test it locally. I may have missed the instructions on how to install it correctly locally and I just used `apt` to install the packages. The problem is that the base repos for ubuntu 18 don't contain the `R` version required (3.5) by the pipeline code, and instead contain the version 3.4. So that's why when I tried to install the requirements for `R`, it failed.

So as a summary, I was running the correct version of Ubuntu but didn't install the correct version of `R`. Didn't know which was the required `R` version until I looked into the `Dockerfile`.
